zerver/tests/test_decorators.py

```diff
@@ -68,367 +59,6 @@
-            "HTTP_USER_AGENT"
-        ] = "ZulipElectron/4.0.3 Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Zulip/4.0.3 Chrome/66.0.3359.181 Electron/3.1.10 Safari/537.36"
-        self.assertEqual(parse_client(req), ("ZulipElectron", "4.0.3"))
-
-        req = HostRequestMock()
-        req.META["HTTP_USER_AGENT"] = "ZulipDesktop/0.4.4 (Mac)"
-        self.assertEqual(parse_client(req), ("ZulipDesktop", "0.4.4"))
-
-        req = HostRequestMock()
-        req.META["HTTP_USER_AGENT"] = "ZulipMobile/26.22.145 (Android 10)"
-        self.assertEqual(parse_client(req), ("ZulipMobile", "26.22.145"))
-
-        req = HostRequestMock()
-        req.META["HTTP_USER_AGENT"] = "ZulipMobile/26.22.145 (iOS 13.3.1)"
-        self.assertEqual(parse_client(req), ("ZulipMobile", "26.22.145"))
-
-        # TODO: This should ideally be Firefox.
-        req = HostRequestMock()
-        req.META[
-            "HTTP_USER_AGENT"
-        ] = "Mozilla/5.0 (X11; Linux x86_64; rv:73.0) Gecko/20100101 Firefox/73.0"
-        self.assertEqual(parse_client(req), ("Mozilla", None))
-
-        # TODO: This should ideally be Chrome.
-        req = HostRequestMock()
-        req.META[
-            "HTTP_USER_AGENT"
-        ] = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.43 Safari/537.36"
-        self.assertEqual(parse_client(req), ("Mozilla", None))
-
-        # TODO: This should ideally be Mobile Safari if we had better user-agent parsing.
-        req = HostRequestMock()
-        req.META[
-            "HTTP_USER_AGENT"
-        ] = "Mozilla/5.0 (Linux; Android 8.0.0; SM-G930F) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.132 Mobile Safari/537.36"
-        self.assertEqual(parse_client(req), ("Mozilla", None))
-
-        post_req_with_client = HostRequestMock()
-        post_req_with_client.POST["client"] = "test_client_1"
-        post_req_with_client.META["HTTP_USER_AGENT"] = "ZulipMobile/26.22.145 (iOS 13.3.1)"
-        self.assertEqual(parse_client(post_req_with_client), ("test_client_1", None))
-
-        get_req_with_client = HostRequestMock()
-        get_req_with_client.GET["client"] = "test_client_2"
-        get_req_with_client.META["HTTP_USER_AGENT"] = "ZulipMobile/26.22.145 (iOS 13.3.1)"
-        self.assertEqual(parse_client(get_req_with_client), ("test_client_2", None))
-
-    def test_unparsable_user_agent(self) -> None:
-        request = HttpRequest()
-        request.POST["param"] = "test"
-        request.META["HTTP_USER_AGENT"] = "mocked should fail"
-        with mock.patch(
-            "zerver.middleware.parse_client", side_effect=JsonableError("message")
-        ) as m, self.assertLogs(level="ERROR"):
-            LogRequests(lambda request: HttpResponse()).process_request(request)
-        request_notes = RequestNotes.get_notes(request)
-        self.assertEqual(request_notes.client_name, "Unparsable")
-        m.assert_called_once()
```

Comment:
I think these first two tests don't really eblong with the rest?

Reply:
Right. Just got that fixed.